src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentLoaderService.cs

```diff
@@ -48,7 +54,35 @@ public PdbSourceDocumentLoaderService()
             return Task.FromResult<TextLoader?>(null);
         }
 
-        private static TextLoader? TryLoadSourceFromDisk(SourceDocument sourceDocument)
+        private static SourceText? TryLoadSourceFromEmbeddedSource(byte[] embeddedTextBytes, Encoding? defaultEncoding)
+        {
+            var uncompressedSize = BitConverter.ToInt32(embeddedTextBytes, 0);
+            var stream = new MemoryStream(embeddedTextBytes, sizeof(int), embeddedTextBytes.Length - sizeof(int));
+
+            if (uncompressedSize != 0)
+            {
+                var decompressed = new MemoryStream(uncompressedSize);
+
+                using (var deflater = new DeflateStream(stream, CompressionMode.Decompress))
+                {
+                    deflater.CopyTo(decompressed);
+                }
+
+                if (decompressed.Length != uncompressedSize)
+                {
+                    return null;
```

Comment:
This is interesting... when do we expect this to happen?

Reply:
If the file is corrupt